exchanges/poloniex/poloniex.go

```
@@ -16,920 +16,1164 @@ import (
+		}
 	}
+	return kline.Interval(0), fmt.Errorf("%w: %q", kline.ErrUnsupportedInterval, interval)
+}
 
-	values.Set("currencyPair", currencyPair)
-	values.Set("rate", strconv.FormatFloat(rate, 'f', -1, 64))
-	values.Set("amount", strconv.FormatFloat(amount, 'f', -1, 64))
-
-	if immediate {
-		values.Set("immediateOrCancel", "1")
+func intervalToString(interval kline.Interval) (string, error) {
+	for x := range supportedIntervals {
+		if supportedIntervals[x].val == interval {
+			return supportedIntervals[x].key, nil
+		}
 	}
+	return "", kline.ErrUnsupportedInterval
+}
 
-	if fillOrKill {
-		values.Set("fillOrKill", "1")
+// WithdrawCurrency withdraws a currency to a specific delegated address
+func (e *Exchange) WithdrawCurrency(ctx context.Context, arg *WithdrawCurrencyRequest) (*Withdraw, error) {
+	if arg.Coin.IsEmpty() {
+		return nil, currency.ErrCurrencyCodeEmpty
 	}
-
-	return result, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, orderType, values, &result)
+	if arg.Amount <= 0 {
+		return nil, limits.ErrAmountBelowMin
+	}
+	if arg.Network == "" {
+		return nil, errInvalidWithdrawalChain
+	}
+	if arg.Address == "" {
+		return nil, errAddressRequired
+	}
+	var resp *Withdraw
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sWithdrawCurrencyEPL, http.MethodPost, "/v2/wallets/withdraw", nil, arg, &resp)
 }
 
-// CancelExistingOrder cancels and order by orderID
-func (e *Exchange) CancelExistingOrder(ctx context.Context, orderID int64) error {
-	result := GenericResponse{}
-	values := url.Values{}
-	values.Set("orderNumber", strconv.FormatInt(orderID, 10))
-
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, poloniexOrderCancel, values, &result)
-	if err != nil {
-		return err
+// GetAccountMargin retrieves account margin information
+func (e *Exchange) GetAccountMargin(ctx context.Context, accountType string) (*AccountMargin, error) {
+	params := url.Values{}
+	if accountType != "" {
```

I can't test this, but documentation says that accountType is required, so this should throw an error if it == "".